firstvoices/backend/views/sites_views.py

```diff
@@ -96,3 +96,76 @@ def list(self, request, *args, **kwargs):
         ]
 
         return Response(data)
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of language sites a given user is a member of, "
+        "grouped by language. If there are no accessible sites or "
+        "the user does not have membership to any site then the "
+        "list will be empty. Sites with no specified language "
+        "will be grouped under 'Other'.",
+        responses={
+            200: inline_serializer(
+                name="InlineLanguageSerializer",
+                fields={
+                    "language": serializers.CharField(),
+                    "sites": SiteSummarySerializer(many=True),
+                },
+            ),
+        },
+    ),
+)
+class MySitesViewSet(
+    AutoPermissionViewSetMixin, mixins.ListModelMixin, viewsets.GenericViewSet
+):
+    """
+    Summary information about language sites a given user has membership to.
+    """
+
+    http_method_names = ["get"]
+    pagination_class = None
+    serializer_class = SiteSummarySerializer
+
+    def get_queryset(self):
+        # get memberships for the user
+        memberships = Membership.objects.filter(user=self.request.user)
+        sites_id_list = []
+
+        # get the site ids from the memberships
+        for membership in memberships:
+            sites_id_list.append(membership.site.id)
+
+        # get the site objects filtered by the id list and order by language
+        queryset = (
+            Site.objects.filter(id__in=sites_id_list)
+            .select_related("language")
+            .order_by(Upper("language__title"), Upper("title"))
+        )
```

Comment:
I think this could all be done in one queryset. Something like Membership.objects.filter(user=x).select_related('site', 'site__language__title').order_by('site__language__title')

Reply:
Good suggestion. Thanks!